### PR TITLE
WiFi support, config file, & version info

### LINKED ISSUE #96: Config File
@lmagder I thought I should ask the expert before doing this just in case you had some particular way to accomplish this.  I already looked up some ways to do it but I wanted your advice before picking one.  I was thinking a .json file with (to start) {launchOnDevice: true, carGPS: true} (those would be the defualts).  
Those 2 things are both highly requested to be disabled but of course not by everyone, so they shouldn't be hardcoded in they should be configurable options. 
Disabling reading the car's GPS did turn out to fix the HUD issue for people and I guess there is a Street Sign Recognition feature where it brings up the street signs on the HUD or something? That was also fixed so there are significant people who would like to be able to have the option to disable AA reading the car's GPS.  And the other thing I just got an email probably the 50th person at least to ask if they can disable AA starting up when they plug in their phone. There are probably a few more things we could put in there too but those 2 are a good start.

### LINKED ISSUE #109: Random Crashes
Hi,
Im using v1.07 and Im getting random crashes while listening to music on Spotify or using Waze, GMaps o even after some calls.

Is there any way I could get some logs to see where the problem could it be and post some of that here?

Regards!

### LINKED ISSUE #110: Google Maps shows car running backwards
Inserting navigation SD card causes your car to turn backwards :). Google Maps shows from now on that you are running backwards as far as you remove the nav card. After removing it the issue disappears and maps behave correctly.

### LINKED ISSUE #118: Reverse camara lost focus  cant find runing process back with fav button
I have been on and off  AA for Mazda, and don't get me wrong thank you very much for your awesome effort. older issues got me out fixes got me in back again  but now i have a reverse camera and  after succesfully conecting to the head unit with the cable conected to my cellphone,   to go out of my parking spot in reverse and then  the AA loses focus. is it me doing something wrong ?

### LINKED ISSUE #121: Android Auto 3's Data-over-WiFi support
The Android Auto 3.0 APK is available for download [at AndroidPolice.com](https://www.androidpolice.com/2018/02/16/android-auto-v3-0-prepares-launch-wi-fi-projection-support-apk-teardown/#Download) and supposedly includes the code required for WiFi support. 

Apparently most cars don't have WiFi built-in like ours do! How exciting would it be for Mazda's unofficial support for AA be better than the competition's official support?

Happy to help in any way that I can (although I am a web developer w/C# mostly, not much experience with C).

### LINKED ISSUE #126: Lost focus when calling someone using Google Assistant
I installed latest version 1.08v downloaded from MazdaTweaks, and Im still having this issue. Also sometimes when someone is calling me and I answere it loses focus too, so then I have to go to the main menu (Mazda Menu) and relaunch Android Auto.

### LINKED ISSUE #129: Once exit back to the main screen, it won't relaunch
Hi,

I'm using v1.08 installed via the All in One Tweak. The AA launches fine 100% when the car starts. However, if I quit the app to the main screen, I can no longer relaunch the AA from the Applications. I got the following errors shown on the screen. I don't know how to extra the log, so took some
img_1582
img_1583
img_1584

Please help. Thanks.

Also great appreciation to your wonderful work.

### LINKED ISSUE #142: Audio focus
Audio focus lost after phone calls. Not always though. Only way to get focus back is to start calling someone, immediately hang up and pressing physical music button.

### LINKED ISSUE #143: Audio stuttering very frequently when navigating with Pixel 3XL
I recently switched to Google Pixel 3XL and noticed that my audio stuttering so bad when navigating with Google Map or Waze. Audio stuttering almost every turns on the map. 

The audio stuttering was much less frequent on my previous phone OG Pixel. Does any Pixel 3 owner experience the same?

Any optimizations we can make on the phone or headunit? I have tried disabled the adaptive battery optimizations on Android Pie without success. I have AA version 1.10+ installed from AIO.

### PR DESCRIPTION
**Pull request by Trevelopment**
🥦
close #96 
close #121 
close #110 
close #129 
close #118 
close #109 
close #126 
close #142 
close #143

### CODE REVIEW

File: mazda/Makefile [hunk 1]
```diff
@@ -74,8 +78,7 @@ clean:
 	rm -f *~ $(TOP)/hu/generated.arm/* dbus/generated_*.* $(OBJS) $(APP) $(DEPS) $(APPLICATION_NAME)_*.zip
 
 install: all
-	cp -a -f headunit installer/config/androidauto/data_persist/dev/bin/
-
+	cp -a -f headunit installer/config/androidauto/data_persist/dev/bin/; cp -a -f headunit.json installer/config/androidauto/data_persist/dev/bin/
```
Comment (Peck07) [inline mazda/Makefile]: this is unnecessary, while the wrapper generates the config file to /tmp/root :)

File: mazda/main.cpp [hunk 2]
```diff
@@ -31,6 +31,13 @@
 #include "command_server.h"
 #include "callbacks.h"
 #include "glib_utils.h"
+#include "config.h"
```
Comment (Peck07) [inline mazda/main.cpp]: one config.h include is enough, you don't need to include/using json as well

File: ubuntu/version.h [hunk 3]
```diff
@@ -0,0 +1 @@
+#define HEADUNIT_VERSION "v1.05-46-g95f1a2d"
```
[the remark this answers is not in the document]
Comment (lmagder) [reply ubuntu/version.h]: Yeah isn't this generated by the makefile or something?

File: mazda/Makefile (hunk 1 above)
Comment (Peck07) [reply mazda/Makefile]: That's true, but I didn't think of updates right now :) But I think there was an idea also, that AIO could generate the config for the user (or just sed the provided default config)

File: ubuntu/version.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (Peck07) [reply ubuntu/version.h]: Yep, it's generated here: 
https://github.com/gartnera/headunit/pull/134/files#diff-2827c073736a637a9ec755528bf1181bR52

File: mazda/installer/tweaks.sh [hunk 4]
```diff
@@ -199,6 +200,7 @@ revert_cmu_files()
             log_message "by reverting changes ... "
             sed -i '/# Android Auto start/d' /jci/scripts/stage_wifi.sh &&
                 sed -i '/headunit-wrapper/d' /jci/scripts/stage_wifi.sh
+                sed -i '/check-usb.sh/d' /jci/scripts/stage_wifi.sh
```
Comment (mishaaq) [inline mazda/installer/tweaks.sh]: You forgot "&&" here, didn't you?

**Pull request merged**